synapse/handlers/room.py

```diff
@@ -346,6 +369,35 @@ def clone_existing_room(
             if old_event:
                 initial_state[k] = old_event.content
 
+        # Resolve the minimum power level required to send any state event
+        # We will give the upgrading user this power level temporarily (if necessary) such that
+        # they are able to copy all of the state events over, then revert them back to their
+        # original power level afterwards
+
+        # Copy over user power levels now as this will not be possible with >100PL users once
+        # the room has been created
+
+        power_levels = initial_state[(EventTypes.PowerLevels, "")]
+
+        # Calculate the minimum power level needed to clone the room
+        event_power_levels = power_levels.get("events", {})
+        state_default = power_levels.get("state_default", 0)
+        ban = power_levels.get("ban")
+        needed_power_level = max(state_default, ban, max(event_power_levels.values()))
+
+        # Raise the requester's power level in the new room if necessary
+        current_power_level = power_levels["users"][requester.user.to_string()]
+        raised_power_level = None
+        if current_power_level < needed_power_level:
+            # Save and return this power level
+            raised_power_level = current_power_level
+
+            # Assign this power level to the requester
+            power_levels["users"][requester.user.to_string()] = needed_power_level
+
+        # Transfer the power levels via power_level_content_override instead of initial_state
+        del initial_state[(EventTypes.PowerLevels, "")]
```

Comment:
We probably want to take a copy of this before mutating it from under the callee?

Reply:
Turns out we don't even need to use the `power_level_content_override` option.